SEP-0012.md

```diff
@@ -0,0 +1,522 @@
+Finally, there is the `.combined_wcs`, this is an APE 14 WCS object which combines the coordinates in the `.wcs` and the `.extra_coords` properties.
+
+In an ideal world, this API would propose using the `.combined_wcs` property as the default for all the WCS operations supported by the `NDCube` API.
+This however, is not currently practical, due to implementation challenges.
+While a robust and performant implementation of `.combined_wcs` is something to be desired, blocking the release of ndcube 2.0 on it would be foolish, therefore this API in its current form makes its use optional.
+
+
+## NDCube
+
+Below is a class definition for the Abstract Base Class (ABC) for `NDCube`.
+Its purpose is to specify the API as completely as possible, including input and return types. 
+The behaviour of all the methods should be described by their included docstrings, a version of this (likely out of sync) has been [PRed to ndcube](https://github.com/sunpy/ndcube/pull/304).
+
+```python
+
+class NDCubeABC(astropy.nddata.NDDataBase):
+
+    @abc.abstractmethod
+    def __init__(self,
+                 data: numpy.typing.ArrayLike,
+                 uncertainty: Any = None,
+                 mask: numpy.typing.ArrayLike = None,
+                 wcs: Union[BaseLowLevelWCS, BaseHighLevelWCS] = None,
+                 meta: Mapping = None,
+                 unit: Union[str, u.Unit] = None,
+                 copy: bool = False):
+        """
+        This signature is inherited from NDData.
+
+        The only difference is that the wcs argument should be considered mandatory.
+        """
+
+    @abc.abstractmethod
+    def __getitem__(self, item: Union[int, slice, Iterable[Union[slice, int]]]) -> "NDCubeABC":
+        """
+        Slice the NDCube object by array index.
+
+        An ndcube object must be sliceable in a similar way to a numpy array,
+        however, support for a `step` is not required and no "fancy indexing" is allowed.
+
+        A step is prohibited due to the abiguity inherent in applying a step to the WCS
+        object; does the step mean a sampling so the coordinate axis becomes discontinuious
+        or does it mean that the size of the pixels should be increased by a factor of the step.
+
+        Fancy indexing, or
+        `advanced indexing <https://numpy.org/doc/stable/reference/arrays.indexing.html#advanced-indexing>`__
+        is not supported both due to similar issues with applying it to the WCS, as
+        well as it being poorly supported outside of numpy arrays.
+
+        When a slice is applied to a NDCube object it must be applied to the following attributes:
+
+          * data
+          * wcs
+          * uncertainty
+          * mask
+          * extra_coords
+
+        the following attributes must be copied unaltered to the returned NDCube:
+
+          * unit
+          * meta
```

Comment:
I think this is too strict in the case of `.meta`.  Slicing of .`meta` should be optional.